pkg/cmd/pulumi/operations/refresh.go

```diff
@@ -321,7 +321,7 @@ func NewRefreshCmd() *cobra.Command {
 	}
 
 	cmd.PersistentFlags().BoolVar(
-		&runProgram, "run-program", false,
+		&runProgram, "run-program", env.RunProgram.Value(),
```

Comment:
I feel like more commonly, we look at the env inside the Cobra callbacks, rather than setting them as defaults here before the command has run. So e.g. this would stay `false`, and inside the function above, we'd do something like `runProgram: runProgram || env.RunProgram.Value()`, maybe? But happy either way as long as we are sure this is safe/robust, just because Cobra is gonna Cobra.

Reply:
That works while false is the default value, but if we want to make --run-program default to true then you need to do this at the cobra/env layer so you can see what is actually set and what's just getting default values.